valhalla/baldr/tilegetter.h

```diff
@@ -45,46 +45,18 @@ class tile_getter_t {
     return false;
   }
 
-  virtual ~tile_getter_t() = default;
-};
-
-/**
- * Default implementation which uses libcurl and curler_pool_t.
- */
-class curl_tile_getter_t : public tile_getter_t {
-public:
   /**
-   * @param pool_size  the number of curler instances in the pool
-   * @param user_agent  user agent to use by curlers for HTTP requests
-   * @param gzipped  whether to request for gzip compressed data
+   * A callback which is called to check if the request should be interrupted.
    */
-  curl_tile_getter_t(const size_t pool_size, const std::string& user_agent, bool gzipped)
-      : curlers_(pool_size, user_agent), gzipped_(gzipped) {
-  }
-
-  using response_t = tile_getter_t::response_t;
+  using interrupt_t = std::function<bool(const std::string&)>;
```

Comment:
the network requests are initiated pretty deep in the call stack. If we want to abort a routing, we don't know which URLs are being fetched at the moment. Not sure if we can find a valid use case for selective cancel.

Reply:
I thought about it but ended up with the idea that having an opportunity to cancel the particular url could be useful and yet could be not usable if we are willing to cancel everything (just don't check the url in this case and return true for everything). The tile getter works with urls and quiet abstract from building routes, searching tiles etc.
But I'm open to this suggestion if you think that would be better. Even more, it would be easier to adapt a function without parameters for cascading cancelation.